web/modules/custom/dbcdk_community/src/Plugin/Block/ProfileQuarantinesBlock.php

```diff
@@ -0,0 +1,213 @@
+   *   The DBCDK Community Service Profile API.
+   * @param \DBCDK\CommunityServices\Api\QuarantineApi $quarantine_api
+   *   The DBCDK Community Service Quarantine API.
+   * @param \Drupal\Core\Datetime\DateFormatter $date_formatter
+   *   Drupal's date formatter to format dates to Drupal Date Formats.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ProfileApi $profile_api, QuarantineApi $quarantine_api, DateFormatter $date_formatter) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->profileApi = $profile_api;
+    $this->quarantineApi = $quarantine_api;
+    $this->dateFormatter = $date_formatter;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('dbcdk_community.api.profile'),
+      $container->get('dbcdk_community.api.quarantine'),
+      $container->get('date.formatter')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $context_username = $this->getContext('username')->getContextData()->getValue();
+    $quarantines = [];
+    try {
+      // Fetch a Community Profile based on the username so we can the
+      // corresponding ID to fetch quarantines.
+      $profile_filter = [
+        'where' => [
+          'username' => $context_username,
+        ],
+      ];
+      /* @var \DBCDK\CommunityServices\Model\Profile */
+      $profile = $this->profileApi->profileFindOne(json_encode($profile_filter));
+
+      // Fetch quarantines.
+      $quarantine_filter = [
+        'order' => 'end DESC',
+        'where' => [
+          'quarantinedId' => $profile->getId(),
+        ],
+      ];
+      /* @var \DBCDK\CommunityServices\Model\Quarantine[] */
+      $quarantines = $this->quarantineApi->quarantineFind(json_encode($quarantine_filter));
+    }
+    catch (ApiException $e) {
+      \Drupal::logger('DBCDK Community Service')->error($e);
+    }
+
+    // Create an array of the fields we wish to display as columns in our table.
+    // The order the fields appear in the array is also the order they will
+    // be displayed.
+    $rows = [];
+    $fields = [
+      'reason' => $this->t('Reason'),
+      'start' => $this->t('Start date'),
+      'end' => $this->t('End date'),
+      'edit_link' => $this->t('Edit'),
+    ];
+    foreach ($quarantines as $quarantine) {
+      $rows[] = $this->parseQuarantine($quarantine, $fields, $context_username);
+    }
+
+    // Build table of quarantines.
+    $build = [];
+    $build['table'] = [
+      '#theme' => 'table',
+      '#header' => $fields,
+      '#rows' => $rows,
+      '#empty' => $this->t('There are no quarantines attached to this profile.', [
+        '%username' => $context_username,
+      ]),
```

Comment:
Good catch - it was a leftover from an initial message that was changed during development :)